src/modules/pico_cyw43/module_pico_cyw43.c

```diff
@@ -1158,12 +1161,81 @@ JERRYXX_FUN(pico_cyw43_wifi_ap_mode) {
 	IP4_ADDR(&gw, 192, 168, 4, 1);
 	IP4_ADDR(&mask, 255, 255, 255, 0);
 
-	static dhcp_server_t dhcp_server;
 	dhcp_server_init(&dhcp_server, &gw, &mask);
 
+  // call callback
+  if (JERRYXX_HAS_ARG(1)) {
+    jerry_value_t callback = JERRYXX_GET_ARG(1);
+    jerry_value_t js_cb = jerry_acquire_value(callback);
+    jerry_value_t errno = jerryxx_get_property_number(
+        JERRYXX_GET_THIS, MSTR_PICO_CYW43_WIFI_ERRNO, 0);
+    jerry_value_t this_val = jerry_create_undefined();
+    jerry_value_t args_p[1] = {errno};
+    jerry_call_function(js_cb, this_val, args_p, 1);
+    jerry_release_value(errno);
+    jerry_release_value(this_val);
+    jerry_release_value(js_cb);
+  }
+
   return jerry_create_undefined();
 }
 
+JERRYXX_FUN(pico_cyw43_wifi_disable_ap_mode) {
+  // verify if AP_mode is enabled
+  int wifi_ap_status = cyw43_tcpip_link_status(&cyw43_state, CYW43_ITF_AP);
+  if (wifi_ap_status != CYW43_LINK_UP) {
+    return jerry_create_error(JERRY_ERROR_COMMON,
+                              (const jerry_char_t *)"WiFi AP_mode is not enabled.");
+  }
+
+  // deinit DHCP server
+  dhcp_server_deinit(&dhcp_server);
+  cyw43_arch_deinit();
+  /* Reset and power up the WL chip */
+  cyw43_hal_pin_low(CYW43_PIN_WL_REG_ON);
+  cyw43_delay_ms(20);
+  cyw43_hal_pin_high(CYW43_PIN_WL_REG_ON);
+  cyw43_delay_ms(50);
+  __cyw43_drv.status_flag = KM_CYW43_STATUS_DISABLED;
+  // init the WiFi chip
+  if (__cyw43_init()) {
+    return jerry_create_error_from_value(create_system_error(EAGAIN), true);
+  }
+  return jerry_create_undefined();
+}
+
+// Function to get the MAC address of connected clients
+JERRYXX_FUN(pico_cyw43_wifi_ap_get_stas) {
+  // verify if AP_mode is enabled
+  int wifi_ap_status = cyw43_tcpip_link_status(&cyw43_state, CYW43_ITF_AP);
+  if (wifi_ap_status != CYW43_LINK_UP) {
+    return jerry_create_error(JERRY_ERROR_COMMON,
+                              (const jerry_char_t *)"WiFi AP_mode is not enabled.");
+  }
+
+  int num_stas, max_stas, MAC_len = 17;
```

Comment:
@jpacora Could you please change the MAC_len to 18? we have a warning message.

/home/jaykim/project/kaluma/kaluma/src/modules/pico_cyw43/module_pico_cyw43.c: In function 'pico_cyw43_wifi_ap_get_stas':
/home/jaykim/project/kaluma/kaluma/src/modules/pico_cyw43/module_pico_cyw43.c:1259:56: warning: 'sprintf' writing a terminating nul past the end of the destination [-Wformat-overflow=]
 1259 |     sprintf(mac_strs[i], "%02x:%02x:%02x:%02x:%02x:%02x", macs[i*6], macs[i*6+1], macs[i*6+2], macs[i*6+3], macs[i*6+4], macs[i*6+5]);
      |                                                        ^
/home/jaykim/project/kaluma/kaluma/src/modules/pico_cyw43/module_pico_cyw43.c:1259:5: note: 'sprintf' output 18 bytes into a destination of size 17
 1259 |     sprintf(mac_strs[i], "%02x:%02x:%02x:%02x:%02x:%02x", macs[i*6], macs[i*6+1], macs[i*6+2], macs[i*6+3], macs[i*6+4], macs[i*6+5]);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~